src/nvs.rs

```diff
@@ -810,28 +814,187 @@ impl RawHandle for EspNvs<NvsEncrypted> {
     }
 }
 
-impl<T: NvsPartitionId> StorageBase for EspNvs<T> {
+impl<T: NvsPartitionId> StorageBase for EspKeyValueStorage<T> {
     type Error = EspError;
 
     fn contains(&self, name: &str) -> Result<bool, Self::Error> {
-        EspNvs::contains(self, name)
+        EspKeyValueStorage::contains(self, name)
     }
 
     fn remove(&mut self, name: &str) -> Result<bool, Self::Error> {
-        EspNvs::remove(self, name)
+        EspKeyValueStorage::remove(self, name)
     }
 }
 
-impl<T: NvsPartitionId> RawStorage for EspNvs<T> {
+impl<T: NvsPartitionId> RawStorage for EspKeyValueStorage<T> {
     fn len(&self, name: &str) -> Result<Option<usize>, Self::Error> {
-        EspNvs::len(self, name)
+        EspKeyValueStorage::len(self, name)
     }
 
     fn get_raw<'a>(&self, name: &str, buf: &'a mut [u8]) -> Result<Option<&'a [u8]>, Self::Error> {
-        EspNvs::get_raw(self, name, buf)
+        EspKeyValueStorage::get_raw(self, name, buf)
     }
 
     fn set_raw(&mut self, name: &str, buf: &[u8]) -> Result<bool, Self::Error> {
-        EspNvs::set_raw(self, name, buf)
+        EspKeyValueStorage::set_raw(self, name, buf)
+    }
+}
+
+impl<T: NvsPartitionId> EspKeyValueStorage<T> {
+    pub fn new(
+        partition: EspNvsPartition<T>,
+        namespace: &str,
+        read_write: bool,
+    ) -> Result<Self, EspError> {
+        Ok(Self(EspNvs::new(partition, namespace, read_write)?))
+    }
+
+    pub fn partition(&self) -> &EspNvsPartition<T> {
+        &self.0 .0
+    }
+
+    pub fn contains(&self, name: &str) -> Result<bool, EspError> {
```

Comment:
Nit: we don't have to do it as it is a bit risky in introducing regressions, but since we have a fully functioning safe `EspNvs` API since quite some time (and now that we have your `find_key` method), I think `EspKeyValueStorage` can be expressed squarely in terms of safe API calls into `EspNvs` without calling the unsafe esp-idf-sys API underneath?

Reply:
Just fix the `u64` serde by restoring the bitshift code. The rest looks quite ok.